packages/lib/slots.ts

```diff
@@ -160,19 +160,47 @@ function buildSlotsWithDateRanges({
   frequency = minimumOfOne(frequency);
   eventLength = minimumOfOne(eventLength);
   offsetStart = offsetStart ? minimumOfOne(offsetStart) : 0;
-
   const slots: { time: Dayjs; userIds?: number[] }[] = [];
   dateRanges.forEach((range) => {
     const startTimeWithMinNotice = dayjs.utc().add(minimumBookingNotice, "minute");
 
     let slotStartTime = range.start.isAfter(startTimeWithMinNotice) ? range.start : startTimeWithMinNotice;
 
-    slotStartTime =
-      slotStartTime.utc().minute() % 15 !== 0
-        ? slotStartTime
-            .startOf("day")
-            .add(slotStartTime.hour() * 60 + Math.ceil(slotStartTime.minute() / 15) * 15, "minute")
-        : slotStartTime;
+    let previousStartTime;
+    // check if we we already have slots on that day (in organizer's timezone)
+    if (
+      slots.length &&
+      dayjs
+        .utc(range.start)
+        .add(range.start.utcOffset())
+        .isSame(dayjs.utc(slots[slots.length - 1].time).add(slots[slots.length - 1].time.utcOffset()), "day")
+    ) {
+      previousStartTime = slots[slots.length - 1].time;
+    }
+
+    if (!previousStartTime) {
```

Comment:
If we don't have a perviousStartTime, meaning there aren't any existing slots already on that day, then we need to find the start time of the first slot